### ISSUE TITLE
Unencrypted messages in a secure session should be more obvious

### ISSUE THREAD

**Issue by pR0Ps** [SMSSecure/SMSSecure#33]
As mentioned [here](https://github.com/SMSSecure/SMSSecure/issues/29#issuecomment-85827342), there's no real warning when a plaintext message is received in the middle of a secure session (other than the message not having a lock icon).

Something more noticeable would be a good idea. I'm thinking some placeholder text like "Unencrypted message, tap to process", where tapping opens an explanation with a cancel/continue button.

**Comment by johanw666** [comment]
Cancel would be quite useless in security terms: the message has already been received by your device so any eavesdroppers would have it already.

Some other color coding might be used for unencrypted messages perhaps: now all received messages are shown in a gray box and sent messages in a green. Perhaps something could be done with that? TextSecure uses blue to differentiate between sms and push, we could expand on that.

**Comment by BastienLQ** [comment]
Yes, a yellow background on unencrypted messages received while a secure session is set up would be fine, plus a warning in the details menu or a badge at the bottom of the message.

**Comment by pR0Ps** [comment]
I'm coming at this from an authenticity point of view, not surveillance. In a secure session it's highly suspicious that an insecure message is received.

Consider:
 - A has a secure session with B
 - C somehow spoofs A's phone number (steals the SIM card, whatever)
 - C sends an unencrypted message to B saying "I'm changing phones, please accept my new key"
 - C (pretending to be A) requests a secure session with B
 - B gets a warning about a different identity key

If B doesn't notice the lack of a padlock on the previous message, they may assume that the message was securely sent from A and be manipulated into accepting the changed key.

(This is yet another reason why you should never accept a new identity key unless you confirm out of band that the request is legitimate).

Anyway, proposed mockup:
![untitled 6](https://cloud.githubusercontent.com/assets/466941/6834924/89369de0-d30d-11e4-81f3-7752afa0ecb8.png)

**Comment by BastienLQ** [comment]
With the triangle, the user could believe the message wasn't sent properly. What about a yellow background (and no triangle)?

**Comment by johanw666** [comment]
Sounds good. If the extra explanatiory text as shown above can be included that would be really good.

**Comment by pR0Ps** [comment]
I'm hesitant to add more colors because that's just another color we have to make work for all themes.

Also, thinking about it now, the amount of text needed to tell the user why the message should be treated with suspicion is too much to put below the message (the message in the mockup is awkward because it had to be short).

I think an unsecured message should be regarded with the same amount of suspicion that a new identity key is. Following that, I'm going back to my original thought where the contents of the message should be unreadable until the user confirms a warning message (by tapping the message).

Thoughts?

This would obviously warn the user and provide enough space to tell them why its a bad thing.

**Comment by BastienLQ** [comment]
I agree. And you could add the triangle in the text to avoid confusion. Or just "WARNING: blah blah" and no triangle.

**Comment by D0L0S** [comment]
What if instead if the triangle a red open padlock is displayed, this should avoid confusion as to weather the message is sent or not aswell as conveying that this is something that the user should be concerned about.
